### PR TITLE
Add shared button components

### PR DESCRIPTION
**Pull request by lyzadanger**
~~Depends on #21~~

Add shared button components, their styles and tests.

The source of the components and the SASS is all c-n-p from the `client` with some very minor changes:

* Adjust paths where needed
* Update exports
* Integrate `variables` changes into SASS structure

### CODE REVIEW

**Comment by lyzadanger** [comment]
p.s. This branch is named poorly; sorry.

File: styles/variables/_colors.scss [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+$grey-6: #3f3f3f;
+
+$grey-7: #202020;
+
+$brand: #bd1c2b;
+
+$link-hover: #84141e !default;
+
+$focus-outline: #51a7e8;
```
Comment (LMS007) [inline styles/variables/_colors.scss:39]: I'm not sure how or why this maps to `color-focus-outline` in focus.scss, but it is. Can you help my understand this?

Comment (lyzadanger) [reply styles/variables/_colors.scss:39]: Yes, in `styles/variables/_index.scss`:

https://github.com/hypothesis/frontend-shared/pull/22/files#diff-8c0e6fb927384efdc1f1df14adbd689d15f2836854e31a320217083e0f224257R1

`@forward 'colors' as color-*;`

It prefixes all of the members coming from the `_colors.scss` partial with the string `color-`

Comment (LMS007) [reply styles/variables/_colors.scss:39]: I see, as we do in the client. Thanks.

**Review by LMS007 — APPROVED** [review]
It seems that styles/components/ is intended to be private and then styles/ and styles/mixins is intended to be the publice faces bits that we export. If that's the case we may need to help clarify the folder intentions so we don't get into a situation where things are a mixed bag of public styles and private in the same folder.  This is not the case yet, but I could see me making that mistake.

GTG beyond my rant!

**Review by esanzgar — APPROVED** [review]
The SCSS is quite sophisticated. I will have to learn SCSS.

**Review by robertknight — APPROVED** [review]
LGTM. I note that @esanzgar opened a couple of related PRs in the client recently which are going to be obsolete due to the components moving to this repo.

**Pull request merged**